storage/merge.go

```diff
@@ -188,35 +187,44 @@ func (l labelGenericQueriers) SplitByHalf() (labelGenericQueriers, labelGenericQ
 // If matchers are specified the returned result set is reduced
 // to label values of metrics matching the matchers.
 func (q *mergeGenericQuerier) LabelValues(ctx context.Context, name string, hints *LabelHints, matchers ...*labels.Matcher) ([]string, annotations.Annotations, error) {
-	res, ws, err := q.lvals(ctx, q.queriers, name, hints, matchers...)
+	res, ws, err := q.mergeResults(q.queriers, hints, func(q LabelQuerier) ([]string, annotations.Annotations, error) {
+		return q.LabelValues(ctx, name, hints, matchers...)
```

Comment:
Maybe this should be `SortedLabelValues()` ?  

(Not something you changed; probably broken by #7448)

Reply:
Isn't `SortedLabelValues` a method on the `tsdb.IndexReader` interface? Its not part of `storage.LabelQuerier`.

Maybe `storage.LabelQuerier` should be changed to always return sorted values.

Reply:
Agreed it is not on `LabelQuerier` and my comment is not directly applicable. 

> Maybe storage.LabelQuerier should be changed to always return sorted values.

Agreed it could be _documented_ like this; we would still have to determine whether every implementation does return sorted results.  I can't figure it out by a quick scan of the code.

Reply:
I created a PR for making this more explicit. 
https://github.com/prometheus/prometheus/pull/14849